sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -400,11 +491,58 @@ case class GpuCast(
     }
 
     withResource(minValue) { minValue =>
-      throwIfAny(values.lessThan(minValue))
+      throwIfAny(inclusiveMin match {
+        case true => values.lessThan(minValue)
+        case false => values.lessOrEqualTo(minValue)
+      })
     }
 
     withResource(maxValue) { maxValue =>
-      throwIfAny(values.greaterThan(maxValue))
+      throwIfAny(inclusiveMax match {
+        case true => values.greaterThan(maxValue)
+        case false => values.greaterOrEqualTo(maxValue)
+      })
+    }
+  }
+
+  /**
+   * Detects outlier values of a column given with specific range, and replaces them with
+   * a inputted substitution value.
+   *
+   * @param values ColumnVector
```

Comment:
nit: ColumnVector is not much of an explanation and does not add anything more than would be auto generated from the signature.

Reply:
Fixed